integrationtests/src/test/java/org/neo4j/bolt/CypherOverBoltIT.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.bolt;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.neo4j.driver.v1.Config;
+import org.neo4j.driver.v1.Driver;
+import org.neo4j.driver.v1.GraphDatabase;
+import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.neo4j.driver.v1.types.Node;
+import org.neo4j.harness.junit.Neo4jRule;
+import org.neo4j.io.fs.FileUtils;
+import org.neo4j.kernel.configuration.Settings;
+import org.neo4j.server.configuration.ServerSettings;
+import org.neo4j.test.rule.SuppressOutput;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class CypherOverBoltIT
+{
+    @Rule
+    public final SuppressOutput suppressOutput = SuppressOutput.suppressAll();
+    @Rule
+    public Neo4jRule graphDb = new Neo4jRule().withConfig( ServerSettings.script_enabled, Settings.TRUE );
+
+    private URL url;
+    private final int lineCountInCSV = 3;
+
+    @Before
+    public void setUp() throws Exception
+    {
+        url = prepareTestImportFile( lineCountInCSV );
+    }
+
+    @Test
+    public void mixingPeriodicCommitAndLoadCSVShouldWork()
+    {
+        try ( Driver driver = GraphDatabase.driver( graphDb.boltURI(), configuration() ); Session session = driver.session() )
+        {
+            StatementResult result = session.run( "USING PERIODIC COMMIT 10000\n" + "LOAD CSV FROM \"" + url + "\" as row fieldterminator \" \"\n" +
```

Comment:
Maybe to test with periodic commit size values that explore the boundaries around `lineCountInCSV` (like `lineCountInCSV - 1`, `lineCountInCSV` and `lineCountInCSV + 1`) is more valuable than 10000.